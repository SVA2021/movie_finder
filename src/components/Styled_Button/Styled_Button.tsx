import {Button, ButtonProps, useTheme} from '@mui/material';

export const StyledButton = (props: ButtonProps) => {
	const theme = useTheme();

	return (
		<Button {...props} sx={{
			borderRadius: theme.shape.borderRadius,
			color: theme.palette.text.secondary,
			"&:hover": {
				color: theme.palette.action.hover,
				boxShadow: `8px 8px 8px ${theme.palette.action.hover},
					-8px -8px 8px ${theme.palette.action.hover}`,
			},
			"&:active": {
				color: theme.palette.action.active,
				boxShadow: `8px 8px 8px ${theme.palette.action.hover},
					-8px -8px 8px ${theme.palette.action.hover}`,
			},
			"&:focus": {
				color: theme.palette.common.white,
				boxShadow: `8px 8px 8px ${theme.palette.action.hover},
					-8px -8px 8px ${theme.palette.action.hover}`,
			},
		}}
		/>
	)
};
StyledButton.muiName = 'Button';