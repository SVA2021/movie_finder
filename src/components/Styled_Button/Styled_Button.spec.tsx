import { fireEvent, render, screen } from '@testing-library/react';
import {StyledButton} from './Styled_Button';

describe('render logo', () => {

	const handleClicker = jest.fn();

	const setUp = () => render(
		<StyledButton variant="contained" aria-label="hello button" size='large' onClick={handleClicker}>Hello World</StyledButton>
	);

	it('snapshot test', () => {
		setUp();
		expect(screen.getByRole('button')).toMatchSnapshot();
	});

	it('find button in document by role', () => {
		setUp();
		expect(screen.getByRole('button')).toBeInTheDocument();
	});

	it('not find button by text', () => {
		setUp();
		expect(screen.queryByText(/another/i)).toBeNull();
	});

	it('find button by text', () => {
		setUp();
		expect(screen.getByText(/hello/i)).toBeInTheDocument();
	});

	it('click button', () => {
		setUp();
		fireEvent.click(screen.getByRole('button'));
		expect(handleClicker).toHaveBeenCalledTimes(1);
	});

	screen.debug();

})