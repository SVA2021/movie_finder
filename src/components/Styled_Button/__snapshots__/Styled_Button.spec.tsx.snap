// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render logo snapshot test 1`] = `
<button
  aria-label="hello button"
  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-9vsaea-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  Hello World
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
